src/centralstore/incremental/RelationBlock.cpp

```diff
@@ -0,0 +1,332 @@
+    }
+    return true;
+}
+
+bool RelationBlock::setPreviousSource(unsigned int newAddress) {
+    if (this->updateRelationRecords(4, newAddress)) {
+        this->source.preRelationId = newAddress;
+
+    } else {
+        throw "Exception: Error while updating the relation previous source address " + std::to_string(newAddress);
+    }
+    return true;
+}
+
+bool RelationBlock::setNextDestination(unsigned int newAddress) {
+    if (this->updateRelationRecords(6, newAddress)) {
+        this->destination.nextRelationId = newAddress;
+    } else {
+        throw "Exception: Error while updating the relation next destination address " + std::to_string(newAddress);
+    }
+    return true;
+}
+
+bool RelationBlock::setPreviousDestination(unsigned int newAddress) {
+    if (this->updateRelationRecords(8, newAddress)) {
+        this->destination.preRelationId = newAddress;
+    } else {
+        throw "Exception: Error while updating the relation previous destination address " + std::to_string(newAddress);
+    }
+    return true;
+}
+
+/**
+ * Update relation record block given the offset to the recored from the begining, i:e
+ *  recordOffset 0 --> Source address
+ *  recordOffset 1 --> Source address
+ *  recordOffset 2 --> Source's next relation block address
+ *  recordOffset 3 --> Source's next relation block partition id
+ *                          .
+ *                          .
+ *                          .
+ * recordOffset 10 --> Relation's property address in the properties DB
+ * */
+bool RelationBlock::updateRelationRecords(int recordOffset, unsigned int data) {
+    int dataOffset = RECORD_SIZE * recordOffset;
+    RelationBlock::relationsDB->seekg(this->addr + dataOffset);
+    if (!RelationBlock::relationsDB->write(reinterpret_cast<char*>(&data), RECORD_SIZE)) {
+        relation_block_logger.error("Error while updating relation data record offset " + std::to_string(recordOffset) +
+                                    "data " + std::to_string(data));
+        return false;
+    }
+    RelationBlock::relationsDB->flush();
+    return true;
+}
+
+bool RelationBlock::isInUse() { return this->usage == '\1'; }
+unsigned int RelationBlock::nextRelationIndex = 1;  // Starting with 1 because of the 0 and '\0' differentiation issue
+
+void RelationBlock::addProperty(std::string name, char* value) {
+    if (this->propertyAddress == 0) {
+        PropertyLink* newLink = PropertyLink::create(name, value);
+        if (newLink) {
+            this->propertyAddress = newLink->blockAddress;
+            // If it was an empty prop link before inserting, Then update the property reference of this node
+            // block
+            this->updateRelationRecords(10, this->propertyAddress);
+        } else {
+            throw "Error occurred while adding a new property link to " + std::to_string(this->addr) + " node block";
+        }
+
```

Comment:
Lets remove empty line

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/115/commits/ed31ba0af33e9629173808960eef3cfe90c5dbe9

Reply:
Done